llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -270,6 +279,17 @@ bool RISCVLegalizerInfo::legalizeCustom(LegalizerHelper &Helper,
     return Helper.lower(MI, 0, /* Unused hint type */ LLT()) ==
            LegalizerHelper::Legalized;
   }
+  case TargetOpcode::G_IS_FPCLASS: {
+    // Turn LLVM IR's floating point classes to that in RISCV,
```

Comment:
If we change the immediate here, would that confuse generic combines that look at G_IS_FPCLASS?

Reply:
Good point. I think right now none of the combiner rules, generic or not, is targeting G_IS_FPCLASS. And I feel like future post-legalizer combiners have to take this in mind.

Reply:
I think we should move the translation to RISC-V encoding to instruction selection.

Reply:
> I think we should move the translation to RISC-V encoding to instruction selection.

Will do!

Reply:
Done